src/s3/explorer/s3FileNode.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as moment from 'moment'
+import * as bytes from 'bytes'
+import * as vscode from 'vscode'
+import { Bucket, DownloadFileRequest, File, S3Client } from '../../shared/clients/s3Client'
+import { ext } from '../../shared/extensionGlobals'
+import { AWSResourceNode } from '../../shared/treeview/nodes/awsResourceNode'
+import { AWSTreeNodeBase } from '../../shared/treeview/nodes/awsTreeNodeBase'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+/**
+ * Represents an object in an S3 bucket.
+ */
+export class S3FileNode extends AWSTreeNodeBase implements AWSResourceNode {
+    public constructor(
+        public readonly bucket: Bucket,
+        public readonly file: File,
+        private readonly s3: S3Client,
+        now: Date = new Date()
+    ) {
+        super(file.name)
+        if (file.sizeBytes !== undefined && file.lastModified) {
+            const readableSize = formatBytes(file.sizeBytes)
+
+            // Prevent clock skew showing future date
```

Comment:
Why would this show a future date?

Reply:
Small variations in server time versus machine time. I observed that as soon as I uploaded something, it would show last updated date as "a few seconds from now". 

I'm sure there are other edge cases where it will show a future date, but this should hopefully squash the most common/annoying one.

Another option would be to clamp the time to be <= now, but I think that would be even more confusing if it was legitimately hitting an edge case. Time is hard.